cli/internal/runcache/runcache.go

```diff
@@ -237,32 +240,29 @@ func (tc TaskCache) OutputWriter() (io.WriteCloser, error) {
 	if err != nil {
 		return nil, err
 	}
-	colorPrefixer := tc.rc.colorCache.PrefixColor(tc.pt.PackageName)
-	prettyTaskPrefix := colorPrefixer(tc.pt.OutputPrefix())
 	bufWriter := bufio.NewWriter(output)
-	if _, err := bufWriter.WriteString(fmt.Sprintf("%s: cache hit, replaying output %s\n", prettyTaskPrefix, ui.Dim(tc.hash))); err != nil {
+
+	cacheHitMessage := tc.cacheHitMessage()
+	if _, err := bufWriter.WriteString(cacheHitMessage); err != nil {
 		// We've already errored, we don't care if there's a further error closing the file we just
 		// failed to write to.
 		_ = output.Close()
 		return nil, err
 	}
+
 	fwc := &fileWriterCloser{
-		file:  output,
-		bufio: bufWriter,
-	}
-	if tc.taskOutputMode == util.NoTaskOutput || tc.taskOutputMode == util.HashTaskOutput {
-		// only write to log file, not to stdout
-		fwc.Writer = bufWriter
-	} else {
-		fwc.Writer = io.MultiWriter(os.Stdout, bufWriter)
+		file:   output,
+		bufio:  bufWriter,
+		Writer: tc.newPrefixWriter(bufWriter),
```

Comment:
I don't think we want a flag to control whether or not we record the prefix. That means later replayers of the logs won't be able to make the decision, it will already be baked in, or not. I think a real solution here is going to have to always skip the prefix / cache hit line when writing the log, and then a flag can control whether or not the prefix and cache hit are displayed during replay / execution.

Reply:
@gsoltis so if I correctly understood, doing the opposite should do the trick? Or you want more flags to display different info (like separate prefix from cache hit)?